src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlConfiguration.java

```diff
@@ -195,6 +197,11 @@ public HtmlConfiguration(Doclet doclet, Locale locale, Reporter reporter) {
 
         conditionalPages = EnumSet.noneOf(ConditionalPage.class);
     }
+    protected void initConfiguration(DocletEnvironment docEnv,
+                                     Function<String, String> resourceKeyMapper) {
+        super.initConfiguration(docEnv, resourceKeyMapper);
+        contents = new Contents(this);
+    }
```

Comment:
I wish initialization of the configuration instance was not that complex... I guess the reason we must do it this way is that used SourceVersion is not known earlier.

Reply:
Agreed that initializing the configuration is overly complex. At least in part, it's because of an earlier desire to use final fields where possible, when using direct access.   And yes, the problem here is that the `SourceVersion` is not known until `init` is called.